fuuk/people/tests/test_admin.py

```diff
@@ -48,3 +48,194 @@ def test_change_view(self):
+        grant = Grant.objects.create(investigator_first_name='John', investigator_last_name='Smith',
+                                     number=1, start=2016, end=2019, agency=self.agency,
+                                     title='Grant', annotation='About the Grant.')
+
+        self.client.login(username=self.username2, password=self.pwd2)
+        response = self.client.get(reverse('admin:people_grant_changelist'))
+        self.assertContains(response, '1 grant')
+        response = self.client.get(reverse('admin:people_grant_change', args=[grant.id]))
+        self.assertContains(response, 'Change')
+
+    def test_change_view_editor(self):
+        self.assertEqual(Grant.objects.count(), 0)
+
+        grant = Grant.objects.create(editor=self.user1, investigator_first_name='John', investigator_last_name='Smith',
+                                     number=1, start=2016, end=2019, agency=self.agency,
+                                     title='Grant', annotation='About the Grant.')
+
+        self.client.login(username=self.username1, password=self.pwd1)
+        response = self.client.get(reverse('admin:people_grant_changelist'))
+        self.assertContains(response, '1 grant')
+        response = self.client.get(reverse('admin:people_grant_change', args=[grant.id]))
+        self.assertContains(response, 'Change')
+        self.client.logout()
+
+    def test_change_view_collaborator_human(self):
+        self.assertEqual(Grant.objects.count(), 0)
+
+        grant = Grant.objects.create(investigator_first_name='John', investigator_last_name='Smith',
+                                     number=1, start=2016, end=2019, agency=self.agency,
+                                     title='Grant', annotation='About the Grant.')
+        GrantCollaborator.objects.create(first_name='Jack', last_name='User1', grant=grant, human=self.human1)
+
+        self.client.login(username=self.username1, password=self.pwd1)
+        response = self.client.get(reverse('admin:people_grant_changelist'))
+        self.assertContains(response, '1 grant')
+        response = self.client.get(reverse('admin:people_grant_change', args=[grant.id]))
+        self.assertContains(response, 'Change')
+        self.client.logout()
+
+    def test_change_view_stranger(self):
+        self.assertEqual(Grant.objects.count(), 0)
+
+        grant = Grant.objects.create(investigator_first_name='John', investigator_last_name='Smith',
+                                     number=1, start=2016, end=2019, agency=self.agency,
+                                     title='Grant', annotation='About the Grant.')
+
+        self.client.login(username=self.username1, password=self.pwd1)
+        response = self.client.get(reverse('admin:people_grant_changelist'))
+        self.assertContains(response, '0 grants')
+        response = self.client.get(reverse('admin:people_grant_change', args=[grant.id]))
+        self.assertEqual(response.status_code, 404)
+        self.client.logout()
```

Comment:
* Some test are still joined together.
 * There's no need to perform `logout` at the end of test.
 * Some functionality is easier to test directly by calling specific methods.